build-tools/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt

```diff
@@ -311,3 +320,233 @@ private fun sshExecutor(project: Project) : ExecutorService = object : ExecutorS
         }
     }
 }
+
+private fun deviceLauncher(project: Project) = object : ExecutorService {
+    private val xcProject = Paths.get(project.testOutputRoot, "launcher")
+
+    private val idb = project.findProperty("idb_path") as? String ?: "idb"
+
+    private val deviceName = project.findProperty("device_name") as? String
+
+    override fun execute(action: Action<in ExecSpec>): ExecResult? {
+        val udid = targetUDID()
+        println("Found device UDID: $udid")
+        install(udid, "build/KonanTestLauncher.ipa")
+        val commands = startDebugServer(udid, "org.jetbrains.kotlin.KonanTestLauncher")
+                .split("\n")
+                .filter { it.isNotBlank() }
+                .flatMap { listOf("-o", it) }
+
+        var savedOut: OutputStream? = null
+        val out = ByteArrayOutputStream()
+        val result = project.exec { execSpec: ExecSpec ->
+            action.execute(execSpec)
+            execSpec.executable = "lldb"
+            execSpec.args = commands + "-b" + "-o" + "command script import ${pythonScript()}" +
+                    "-o" + ("process launch" +
+                        (execSpec.args.takeUnless { it.isEmpty() }
+                                ?.let { " -- ${it.joinToString(" ")}" }
+                                ?: "")) +
+                    "-o" + "get_exit_code" +
+                    "-k" + "get_exit_code" +
+                    "-k" + "exit -1"
+            savedOut = execSpec.standardOutput
+            execSpec.standardOutput = out
+        }
+        out.toString()
+                .also { if (project.verboseTest) println(it) }
+                .split("\n")
+                .run { drop(indexOfFirst { s -> s.startsWith("(lldb) process launch") } + 1) }
```

Comment:
Do we really need this `run { drop(...)}` here? Why not just `drop(...)`?

It seems that we also can replace `drop(indexOfFirst { ... } + 1)` with `dropWhile { !it.startsWith(...) }.drop(1)` but I'm not sure that it would be more readable.